### PR TITLE
Fix account flow validators and update FETextField to support hints, dynamic link fix, UITableView dynamic height, fix forgot password worker, RootNavigationController color updates, cleanup

### PR DESCRIPTION
**Pull request by kenanmamedoff**

### CODE REVIEW

File: brd-ios/breadwallet/src/Redesign/Scenes/ForgotPassword/ForgotPasswordInteractor.swift [hunk 1]
```diff
@@ -28,9 +28,19 @@ class ForgotPasswordInteractor: NSObject, Interactor, ForgotPasswordViewActions
...
         }
         
-        let isEmailValid = dataStore?.email?.isValidEmailAddress ?? false
+        let isEmailValid = dataStore?.email.isValidEmailAddress ?? false
+        let isEmailEmpty = dataStore?.email.isEmpty == true
+        var emailState: DisplayState = .selected
+        if isEmailEmpty {
+            emailState = .selected
+        } else {
+            emailState = isEmailValid ? .selected : .error
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/ForgotPassword/ForgotPasswordInteractor.swift:37]: You already have default state set to `.selected`, maybe just change it if `!isEmailEmpty && !isEmailValid`?

File: brd-ios/breadwallet/src/Redesign/Scenes/ForgotPassword/ForgotPasswordPresenter.swift [hunk 2]
```diff
@@ -35,7 +35,15 @@ final class ForgotPasswordPresenter: NSObject, Presenter, ForgotPasswordActionRe
...
     
     func presentValidate(actionResponse: ForgotPasswordModels.Validate.ActionResponse) {
-        viewController?.displayValidate(responseDisplay: .init(isValid: actionResponse.isValid))
+        let isValid = actionResponse.isEmailValid
+        
+        viewController?.displayValidate(responseDisplay: .init(email: actionResponse.email,
+                                                               isEmailValid: actionResponse.isEmailValid,
+                                                               isEmailEmpty: actionResponse.isEmailEmpty,
+                                                               emailModel: .init(title: L10n.Account.email,
+                                                                                 hint: actionResponse.emailState == .error ? "Wrong e-mail address." : nil,
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/ForgotPassword/ForgotPasswordPresenter.swift:44]: Localize hint?

File: brd-ios/breadwallet/src/Redesign/Scenes/SignIn/SignInInteractor.swift [hunk 3]
```diff
@@ -32,10 +32,32 @@ class SignInInteractor: NSObject, Interactor, SignInViewActions {
...
-        let isValid = FieldValidator.validate(fields: [dataStore?.email,
-                                                       dataStore?.password])
+        let isEmailValid = dataStore?.email.isValidEmailAddress ?? false
+        let isEmailEmpty = dataStore?.email.isEmpty == true
+        var emailState: DisplayState = .selected
+        if isEmailEmpty {
+            emailState = .selected
+        } else {
+            emailState = isEmailValid ? .selected : .error
+        }
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignIn/SignInInteractor.swift:42]: Same comment as above

File: brd-ios/breadwallet/src/Redesign/Scenes/SignIn/SignInInteractor.swift [hunk 4]
```diff
@@ -32,10 +32,32 @@ class SignInInteractor: NSObject, Interactor, SignInViewActions {
...
+        
+        let isPasswordValid = dataStore?.password.isValidPassword ?? false
+        let isPasswordEmpty = dataStore?.password.isEmpty == true
+        var passwordState: DisplayState = .selected
+        if isPasswordEmpty {
+            passwordState = .selected
+        } else {
+            passwordState = isPasswordValid ? .selected : .error
+        }
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignIn/SignInInteractor.swift:51]: Ditto

File: brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpInteractor.swift [hunk 5]
```diff
@@ -38,14 +38,50 @@ class SignUpInteractor: NSObject, Interactor, SignUpViewActions {
...
+        }
+        
+        let isPasswordValid = dataStore?.password.isValidPassword ?? false
+        let isPasswordEmpty = dataStore?.password.isEmpty == true
+        var passwordState: DisplayState = .selected
+        if isPasswordEmpty {
+            passwordState = .selected
+        } else {
+            passwordState = isPasswordValid ? .selected : .error
+        }
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpInteractor.swift:57]: Also here

File: brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpInteractor.swift [hunk 6]
```diff
@@ -38,14 +38,50 @@ class SignUpInteractor: NSObject, Interactor, SignUpViewActions {
...
+        
+        let isPasswordAgainValid = dataStore?.passwordAgain.isValidPassword ?? false
+        let isPasswordAgainEmpty = dataStore?.passwordAgain.isEmpty == true
+        var passwordAgainState: DisplayState = .selected
+        if isPasswordAgainEmpty {
+            passwordAgainState = .selected
+        } else {
+            passwordAgainState = isPasswordAgainValid ? .selected : .error
+        }
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpInteractor.swift:66]: And here

File: brd-ios/breadwallet/src/Redesign/Scenes/ForgotPassword/ForgotPasswordPresenter.swift (hunk 2 above)
Comment (kenanmamedoff) [reply brd-ios/breadwallet/src/Redesign/Scenes/ForgotPassword/ForgotPasswordPresenter.swift:44]: Have a ticket for this.

**Review by dangelovska — APPROVED** [review]

File: brd-ios/breadwallet/src/Redesign/Scenes/REUSABLE/BaseInfo/BaseInfoViewController.swift [hunk 7]
```diff
@@ -16,15 +16,15 @@ class BaseInfoViewController: BaseTableViewController<BaseCoordinator,
     typealias Models = BaseInfoModels
     
     var imageName: String? { return Asset.timeoutStatusIcon.name }
-    var titleText: String? { return "OVERRIDE IN SUBCLASS" }
-    var descriptionText: String? { return "THIS AS WELL" }
+    var titleText: String? { return "Override" }
+    var descriptionText: String? { return "Override" }
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/REUSABLE/BaseInfo/BaseInfoViewController.swift:20]: Should this be localized?

File: brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift [hunk 8]
```diff
@@ -44,9 +44,38 @@ final class SignUpPresenter: NSObject, Presenter, SignUpActionResponses {
...
+        let noticeConfiguration = LabelConfiguration(font: Fonts.Body.three, textColor: textColor)
+        
+        viewController?.displayValidate(responseDisplay:
+                .init(email: actionResponse.email,
+                      password: actionResponse.password,
+                      passwordAgain: actionResponse.passwordAgain,
+                      isEmailValid: actionResponse.isEmailValid,
+                      isEmailEmpty: actionResponse.isEmailEmpty,
+                      emailModel: .init(title: L10n.Account.enterEmail,
+                                        hint: actionResponse.emailState == .error ? "Wrong e-mail address." : nil,
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift:64]: Should this be localized?

File: brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift [hunk 9]
```diff
@@ -44,9 +44,38 @@ final class SignUpPresenter: NSObject, Presenter, SignUpActionResponses {
...
+                                        displayState: actionResponse.emailState),
+                      isPasswordValid: actionResponse.isPasswordValid,
+                      isPasswordEmpty: actionResponse.isPasswordEmpty,
+                      passwordModel: .init(title: L10n.Account.enterPassword,
+                                           displayState: actionResponse.passwordState),
+                      isPasswordAgainValid: actionResponse.isPasswordAgainValid,
+                      isPasswordAgainEmpty: actionResponse.isPasswordAgainEmpty,
+                      passwordAgainModel: .init(title: L10n.Account.confirmPassword,
+                                                hint: actionResponse.passwordAgainState == .error
+                                                && !actionResponse.passwordsMatch ? "Passwords should match." : nil,
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift:74]: Ditto

Comment (kenanmamedoff) [reply brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift:74]: image

**Pull request merged**